src/mongoose_http_client.erl

```diff
@@ -13,136 +13,123 @@
-%% internal functions
-
-make_pool(Name, Opts) ->
-    #pool{name = Name,
-          server = gen_mod:get_opt(server, Opts, "http://localhost"),
-          size = gen_mod:get_opt(pool_size, Opts, 20),
-          max_overflow = gen_mod:get_opt(max_overflow, Opts, 5),
-          path_prefix = list_to_binary(gen_mod:get_opt(path_prefix, Opts, "/")),
-          pool_timeout = gen_mod:get_opt(pool_timeout, Opts, 200),
-          request_timeout = gen_mod:get_opt(request_timeout, Opts, 2000)}.
-
-do_start_pool(#pool{name = Name,
-                    server = Server,
-                    size = PoolSize,
-                    max_overflow = MaxOverflow}) ->
-    ProcName = pool_proc_name(Name),
-    PoolOpts = [{name, {local, ProcName}},
-                {size, PoolSize},
-                {max_overflow, MaxOverflow},
-                {worker_module, mongoose_http_client_worker}],
-    {ok, _} = supervisor:start_child(
-                sup_proc_name(),
-                poolboy:child_spec(ProcName, PoolOpts, [Server, []])),
-    ok.
-
-pool_proc_name(PoolName) ->
-    list_to_atom("mongoose_http_client_pool_" ++ atom_to_list(PoolName)).
-
-sup_proc_name() ->
-    mongoose_http_client_sup.
-
-tab_name() ->
-    mongoose_http_client_pools.
+do_start_pool(PoolName, Opts) ->
+    SelectionStrategy = gen_mod:get_opt(selection_strategy, Opts, available_worker),
+    PathPrefix = list_to_binary(gen_mod:get_opt(path_prefix, Opts, "/")),
+    RequestTimeout = gen_mod:get_opt(request_timeout, Opts, 2000),
+    PoolTimeout = gen_mod:get_opt(pool_timeout, Opts, 5000),
+    PoolSettings = #mongoose_worker_pool{selection_strategy = SelectionStrategy,
+                                         extra = PathPrefix,
+                                         request_timeout = RequestTimeout,
+                                         pool_timeout = PoolTimeout},
+    mongoose_wpool:save_pool_settings(PoolName, PoolSettings),
+    PoolSize = gen_mod:get_opt(pool_size, Opts, 20),
+    Server = gen_mod:get_opt(server, Opts),
+    HttpOpts = gen_mod:get_opt(http_opts, Opts, []),
+    PoolOpts = [{workers, PoolSize}, {worker, {fusco, {Server, HttpOpts}}}
+                | gen_mod:get_opt(pool_opts, Opts, [])],
+    wpool:start_sup_pool(PoolName, PoolOpts).
 
 make_request(Pool, Path, Method, Headers, Query) ->
-    #pool{path_prefix = PathPrefix,
-          name = PoolName,
-          pool_timeout = PoolTimeout,
-          request_timeout = RequestTimeout} = Pool,
+    case catch pool_name(Pool) of
+        {'EXIT', badarg, _} ->
+            {error, pool_not_started};
+        PoolName ->
+            case mongoose_wpool:get_pool_settings(PoolName) of
+                undefined ->
+                    {error, pool_not_started};
+                PoolOpts ->
+                    make_request(PoolName, PoolOpts, Path, Method, Headers, Query)
+            end
+    end.
+
+make_request(PoolName, PoolOpts, Path, Method, Headers, Query) ->
+    #mongoose_worker_pool{extra = PathPrefix,
```

Comment:
Such details should be hidden behind `mongoose_wpool` API, e.g. `mongoose_wpool:call`.